js/main.js

```diff
@@ -1,2 +1,23 @@
 /* global data */
 /* exported data */
+var photo = document.querySelector('.image');
+var pUrl = document.querySelector('.photo-url');
+pUrl.addEventListener('input', function (event) {
+  photo.src = event.target.value;
+});
+
+var formSelector = document.querySelector('.main-form');
+formSelector.addEventListener('submit', function (event) {
+  event.preventDefault();
+  var values = {
+    title: formSelector.elements.title.value,
+    url: formSelector.elements.url.value,
+    textbox: formSelector.elements.notes.value,
+    entryId: data.nextEntryId
+  };
+  data.nextEntryId++;
+  data.entries.push(values);
```

Comment:
You should be adding to the beginning of the array, not the end. You will need to use `unshift` here.